agent/engine/docker_task_engine.go

```diff
@@ -1874,7 +1875,35 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	// Update the environment variables FLUENT_HOST and FLUENT_PORT depending on the supported network modes - bridge
 	// and awsvpc. For reference - https://docs.docker.com/config/containers/logging/fluentd/.
 	if hostConfig.LogConfig.Type == logDriverTypeFirelens {
-		hostConfig.LogConfig = getFirelensLogConfig(task, container, hostConfig, engine.cfg)
+		// We need the Docker server version in order to generate the appropriate firelens log config
+		dockerServerVersion, err := engine.Version()
+		if err != nil {
+			logger.Error("Failed to determine Docker server version for Firelens log config generation", logger.Fields{
+				field.TaskID:    task.GetID(),
+				field.Container: container.Name,
+				field.Error:     err,
+			})
+			return dockerapi.DockerContainerMetadata{
+				Error: dockerapi.CannotCreateContainerError{FromError: errors.Wrapf(versionErr,
+					"failed to create container - container uses awsfirelens log driver and we failed to "+
+						"determine the Docker server version for Firelens log config generation")},
+			}
```

Comment:
Will we need to update any docs to cover this error?

Reply:
Good question, let me check that

Reply:
Currently, our troubleshooting documentation does not have any CannotCreateContainer errors documented.
I will raise this request with the docs team.